File: go/alpha_number.go
// @Since 2024-03-21.
// @Author Fury, All rights Reserved.

package validator

const alphaNumberRegexp string = "^[a-zA-Z][a-zA-Z0-9]*$"

// IsValidAlphaNumber 字母数字标识符
// Validate is the value a valid alpha number
func IsValidAlphaNumber(value string, allowBlank bool) bool {
	if isBlank(value) {
		return allowBlank
	}

	return regexp.MustCompile(alphaNumberRegexp).MatchString(value)
}

File: java/src/main/java/cn/xlibs/lib4j/validator/annotation/AlphaNumber.java
package cn.xlibs.lib4j.validator.annotation;

import cn.xlibs.lib4j.validator.validation.AlphaNumberValidator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

/**
 * 字母数字标识
 * Alpha Number Validator
 *
 * @author Fury
 * @since 2024-03-21
 * <p>
 * All rights Reserved.
 */
@Documented
@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = AlphaNumberValidator.class)
public @interface AlphaNumber {
    boolean required() default true;

    String message() default "参数格式错误";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}
